tests/unittests/test_sync.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+from unittest import mock
+from parameterized import parameterized
+from tap_quickbooks.client import QuickbooksClient
+from tap_quickbooks.sync import do_sync
+from singer import utils
+
+class Schema:
+    """Mocked Schema"""
+
+    def __init__(self, stream_name) -> None:
+        self.stream = stream_name
+
+    def to_dict(self):
+        return {'stream': self.stream}
+
+class Stream:
+    """Mocked Stream """
+
+    def __init__(self, stream_name) -> None:
+        self.tap_stream_id = stream_name
+        self.schema = Schema(stream_name)
+        self.metadata = {}
+
+class MockCatalog:
+    """Mocked Catalog """
+
+    def __init__(self, stream) -> None:
+        self.stream = stream
+
+    def get_selected_streams(self, state):
+        for stream in self.stream:
+            yield Stream(stream)
+
+def get_data(stream_name, count: int = 0):
+    """Return data based on stream_name and count of data """
+
+    if stream_name == 'accounts':
+        return {'QueryResponse': {'Account': [{'data': 'value', 'MetaData': {'LastUpdatedTime': '2012-08-21T00:00:00Z'}}]}}
+
+    if stream_name == "profit_loss_report":
+        return [{"ReportDate": "2022-06-21T00:00:00.000000Z", "AccountingMethod": "Accrual"}]
+
+    if stream_name == 'deleted_objects':
+        mock_data = {'CDCResponse': [{'QueryResponse': [{'Account': []}]}]}
+        for i in range(count):
+            dummy = {'Id': i+1, 'status': 'Deleted',
+                     'MetaData': {'LastUpdatedTime': '2022-06-21T00:00:00Z'}}
+            mock_data.get('CDCResponse')[0].get('QueryResponse')[
+                0].get('Account').append(dummy)
+        return mock_data
+
+class TestSyncCov(unittest.TestCase):
+
+    @mock.patch('tap_quickbooks.client.QuickbooksClient.get')
+    @mock.patch('singer.write_record')
+    @mock.patch('singer.write_state')
+    def test_sync_accounts(self, mock_write_state, mock_write_rcord, mock_get):
```

Comment:
Is this the correct order for the decorators? If so, is there a way to make this more clear? each test function takes arguments in the opposite order of the decorators

Reply:
Please refer to the above response.